src/allmydata/web/status.py

```diff
@@ -759,114 +759,162 @@ def render_server_timings(self, ctx, data):
         return T.li["Per-Server Fetch Response Times: ", l]
 
 
-class PublishStatusPage(rend.Page, RateAndTimeMixin):
-    docFactory = getxmlfile("publish-status.xhtml")
+# Renders "status/publish-%n"
+class PublishStatusPage(MultiFormatResource):
 
-    def __init__(self, data):
-        rend.Page.__init__(self, data)
-        self.publish_status = data
+    def __init__(self, publish_status):
+        """
+        :publish_status mutable.publish.PublishStatus: publish stats provider.
```

Comment:
`:param mutable.publish.PublishStatus publish_status: ...`

Reply:
Updated in ebe8022.

If only pydocstyle wasn't so noisy... :-)